src/families/stellar/bridge/libcore.js

```diff
@@ -0,0 +1,265 @@
+import { scanAccounts } from "../../../libcore/scanAccounts";
+import { getAccountNetworkInfo } from "../../../libcore/getAccountNetworkInfo";
+import { sync } from "../../../libcore/syncAccount";
+import broadcast from "../libcore-broadcast";
+import signOperation from "../libcore-signOperation";
+import { withLibcore } from "../../../libcore/access";
+import memoTypeCheck from "../memo-type-check";
+
+import { getWalletName } from "../../../account";
+import { getOrCreateWallet } from "../../../libcore/getOrCreateWallet";
+import { getCoreAccount } from "../../../libcore/getCoreAccount";
+
+const checkRecipientExist = (account, recipient) =>
+  withLibcore(async core => {
+    const { derivationMode, currency } = account;
+
+    const walletName = getWalletName(account);
+
+    const coreWallet = await getOrCreateWallet({
+      core,
+      walletName,
+      currency,
+      derivationMode
+    });
+
+    const stellarLikeWallet = await coreWallet.asStellarLikeWallet();
+    const recipientExist = await stellarLikeWallet.exists(recipient);
+
+    return recipientExist;
+  });
+
+const createTransaction = () => ({
+  family: "stellar",
+  amount: BigNumber(0),
+  baseReserve: null,
+  networkInfo: null,
+  fees: null,
+  recipient: "",
+  memoValue: null,
+  memoType: null,
+  useAllAmount: false,
+  memoTypeRecommended: null
+});
+
+const updateTransaction = (t, patch) => {
+  if ("recipient" in patch && patch.recipient !== t.recipient) {
+    return { ...t, ...patch, memoType: null };
+  }
+  return { ...t, ...patch };
+};
+
+const isMemoValid = (memoType: string, memoValue: string): boolean => {
+  switch (memoType) {
+    case "MEMO_TEXT":
+      if (memoValue.length > 28) {
+        return false;
+      }
+      break;
+
+    case "MEMO_ID":
+      if (BigNumber(memoValue.toString()).isNaN()) {
+        return false;
+      }
+      break;
+
+    case "MEMO_HASH":
+    case "MEMO_RETURN":
+      if (!memoValue.length || memoValue.length !== 32) {
+        return false;
+      }
+      break;
+  }
+  return true;
+};
+
+const isAccountIsMultiSign = async account =>
+  withLibcore(async core => {
+    const { coreAccount } = await getCoreAccount(core, account);
+
+    const stellarLikeAccount = await coreAccount.asStellarLikeAccount();
+    const signers = await stellarLikeAccount.getSigners();
+
+    return signers.length > 1;
+  });
+
+const getTransactionStatus = async (a, t) => {
+  const errors = {};
+  const warnings = {};
+  const useAllAmount = !!t.useAllAmount;
+
+  if (a.freshAddress === t.recipient) {
+    errors.recipient = new InvalidAddressBecauseDestinationIsAlsoSource();
+  } else {
+    const { recipientError, recipientWarning } = await validateRecipient(
+      a.currency,
+      t.recipient
+    );
+
+    if (recipientError) {
+      errors.recipient = recipientError;
+    }
+
+    if (recipientWarning) {
+      warnings.recipient = recipientWarning;
+    }
+  }
+
+  if (await isAccountIsMultiSign(a)) {
```

Comment:
I think it's efficient enough, but a cache will be super nice !

Reply:
Hmmm on a second thought, may be not.
It will be a rare case but it will definitely be a problem if we cache it and a user that just authorize multisign, immediately try to do a send.

Reply:
makes sense